### PR TITLE
chore: bump docs-sidenav to fix .io mobile nav

### PR DESCRIPTION
**Pull request by BRKalow**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

Fix broken mobile docs nav on .io sites

### CODE REVIEW

File: src/components/open-api-page/index.tsx [hunk 1]
```diff
@@ -65,7 +64,6 @@ function OpenApiPage({
 			/>
 			<DocsSidenav
 				product={productSlug}
-				Link={Link}
```
Comment (BRKalow) [inline src/components/open-api-page/index.tsx:68]: No loner a necessary prop.

File: src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts [hunk 2]
```diff
@@ -17,8 +17,7 @@ interface GetPathBreadcrumbsOpts {
 	/**
 	 * An array of nav nodes, as defined by our sidenav component.
 	 */
-
-	navData: NavData
+	navData: MenuItem[]
```
Comment (BRKalow) [inline src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts:20]: It seems more correct to use our `MenuItem` interface here, plus it fixed some type errors from type updates in `docs-sidenav` 😄

File: src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { VersionSelectItem } from '@hashicorp/react-docs-page/server/loaders/remote-content'
-import { NavData } from '@hashicorp/react-docs-sidenav'
+import { MenuItem } from 'components/sidebar'
```
Comment (thiskevinwang) [inline src/components/breadcrumb-bar/utils/get-docs-breadcrumbs.ts:2]: tiny nit
```suggestion
import { type MenuItem } from 'components/sidebar'
```

**Review by thiskevinwang — APPROVED** [review]
Looks like only a major package bump and type-fixes
👍

**Pull request merged**